### DISCUSSION TITLE
Routine creation

### DISCUSSION THREAD

**Discussion by Labowskyy** [dmroeder/pylogix#246] [Q&A] [answered]
Hello,

I'm looking to have ascript that will create routines and logic along with the tags required. I was wondering where to get started with something like this? I looked through the examples and while those were great they seemed to relate more to tags where I'm unsure how to start creating the logic.

Thank you

**Comment by TheFern2** [comment, marked as answer]
Hi there, what you're trying to do is beyond the scope of pylogix. What you might be able to do is import export the l5x file and manipulate the file which is an xml but you're on your own for creating logic and routines. I don't know of any libraries which do what you want to do.

**Reply by Labowskyy** [reply]
Thanks for the reply, that was my thoughts aswell. I was hoping to have something a little more sophisticated than just spitting out L5X files but this looks like the route.

Thank you!

**Comment by dmalicke71** [comment]
I have used Excel and VBA scripts to create tags, tag descriptions and ladder logic routines in xml format. You can export a program and use that to get the required file format and build files that you can import. I can post an example if you want when I get to the office.  I used it the build boiler plate code like I/O mapping, alarming, etc. and all the required tags. I also had scripts to build WonderWare tags and tags for the simulator. Dustin M.Sent from my iPadOn Apr 24, 2024, at 4:54 PM, Labowskyy ***@***.***> wrote:﻿
Thanks for the reply, that was my thoughts aswell. I was hoping to have something a little more sophisticated than just spitting out L5X files but this looks like the route.
Thank you!

**Reply by dmroeder** [reply]
Maybe this module will help?

https://github.com/jvalenzuela/l5x

**Reply by Labowskyy** [reply]
@dmalicke71 I wouldn't mind to look at an example just for reference as I was originally going to do it that way.

@dmroeder Thank you for this package, it looks like a more intuitive way to handle specific changes instead of just straight XML.

Thanks again for the help!

**Comment by dmroeder** [comment]
Is uhhh, your username a reference to something @Labowskyy?

**Reply by Labowskyy** [reply]
Haha yeah it is, the movie the big lebowski. 

I love the movie but the name was taken originally when I made it on some platform a while ago so I've just stuck with this iteration, without the extra y, ever since.

**Reply by dmroeder** [reply]
I almost posted, "Ok sir, you're a Labowskyy, I'm a Labowskyy, that's terrific."

**Reply by TheFern2** [reply]
8 year olds dude
